lvgl/src/display.rs

```diff
@@ -0,0 +1,268 @@
+use crate::functions::CoreError;
+use crate::{disp_drv_register, disp_get_default, get_str_act};
+use crate::{Box, RunOnce};
+use crate::{Color, Obj};
+use core::cell::RefCell;
+use core::mem::MaybeUninit;
+use core::ptr::NonNull;
+use core::{ptr, result};
+use parking_lot::const_mutex;
+use parking_lot::Mutex;
+
+pub const DISP_HOR_RES: usize = lvgl_sys::LV_HOR_RES_MAX as usize;
+pub const DISP_VER_RES: usize = lvgl_sys::LV_VER_RES_MAX as usize;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum DisplayError {
+    NotAvailable,
+    FailedToRegister,
+    NotRegistered,
+}
+
+type Result<T> = result::Result<T, DisplayError>;
+
+pub struct Display {
+    pub(crate) disp: NonNull<lvgl_sys::lv_disp_t>,
+}
+
+impl Display {
+    pub(crate) fn from_raw(disp: NonNull<lvgl_sys::lv_disp_t>) -> Self {
+        Self { disp }
+    }
+
+    pub fn register<F, const N: usize>(
+        draw_buffer: &'static DrawBuffer<N>,
+        display_update: F,
+    ) -> Result<Self>
+    where
+        F: FnMut(&DisplayRefresh<N>) + 'static,
+    {
+        let mut display_diver = DisplayDriver::new(draw_buffer, display_update)?;
+        Ok(disp_drv_register(&mut display_diver)?)
+    }
+
+    pub fn get_scr_act(&self) -> Result<Obj> {
+        Ok(get_str_act(Some(&self))?)
+    }
+}
+
+impl Default for Display {
+    fn default() -> Self {
+        disp_get_default().expect("LVGL must be INITIALIZED")
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct DefaultDisplay {}
+
+impl DefaultDisplay {
+    /// Gets the screen active of the default display.
+    pub fn get_scr_act() -> Result<Obj> {
+        Ok(get_str_act(None)?)
+    }
+}
+
+pub struct DrawBuffer<const N: usize> {
+    initialized: RunOnce,
+    refresh_buffer: Mutex<RefCell<[MaybeUninit<lvgl_sys::lv_color_t>; N]>>,
```

Comment:
I'm using a Mutex here just to make this `Sync`, not sure if there is a way to remove it but still keep the memory statically allocated.